files/image_config/process-reboot-cause/process-reboot-cause

```diff
@@ -100,6 +108,12 @@ def main():
             previous_reboot_cause = cause_file.readline().rstrip('\n')
             cause_file.close()
 
+        # If it is FirstTime Boot and previous_reboot_cause is unknown
+        # Update the reboot cause as required
+        if os.path.isfile(FIRST_BOOT_PLATFORM_FILE):
+            if (previous_reboot_cause == UNKNOWN_REBOOT_CAUSE):
+                 previous_reboot_cause = UNKNOWN_REBOOT_CAUSE
```

Comment:
Hi @sujinmkang 

Ideally the code can be 

       # Update the reboot cause as required
       if os.path.isfile(FIRST_BOOT_PLATFORM_FILE):
              if (previous_reboot_cause == UNKNOWN_REBOOT_CAUSE):
                       previous_reboot_cause = "First Boot"

When the device is booted with sonic first time,  the first boot file will be present.

Now when we install Sonic the first time (lets say FTOS to sonic conversion), 
the show reboot-cause will return "Unable to determine the previous reboot cause". 
Instead it will be good if we return clear data like "First Time Boot"  or "FirstDeployment"
However this has to be decided by MSFT. Hence i set it as previous value itself.
MSFT, if needed can change to relevant return string. 

Thanks
Sridhar.R

Reply:
@sridhar-ravindran I got it. Let's keep it as is for now. We can revisit if needed.

Reply:
Can you please resolve the merge conflicts ?

Reply:
Hi @sujinmkang 
Synced to latest and resolved conflicts